### ISSUE TITLE
 Mismatched Domain types: varchar(30) vs varchar

### ISSUE THREAD

**Issue by antw12** [prestodb/presto#10957]
```
com.facebook.presto.spi.PrestoException: Error opening Hive split hdfs://p01001532067275.cern.ch/user/hive/warehouse/tpcds_10000_parquet.db/customer_address/9f4942e27c4cb312-966d8e0f00000006_1628803403_data.0.parq (offset=0, length=9579444): Mismatched Domain types: varchar(30) vs varchar
	at com.facebook.presto.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:227)
	at com.facebook.presto.hive.parquet.ParquetPageSourceFactory.createPageSource(ParquetPageSourceFactory.java:122)
	at com.facebook.presto.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:157)
	at com.facebook.presto.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:93)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:44)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:56)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:216)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:373)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:282)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:276)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:973)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:492)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Mismatched Domain types: varchar(30) vs varchar
	at com.facebook.presto.spi.predicate.Domain.checkCompatibility(Domain.java:229)
	at com.facebook.presto.spi.predicate.Domain.intersect(Domain.java:184)
	at com.facebook.presto.spi.predicate.TupleDomain.intersect(TupleDomain.java:197)
	at com.facebook.presto.spi.predicate.TupleDomain.overlaps(TupleDomain.java:300)
	at com.facebook.presto.hive.parquet.predicate.TupleDomainParquetPredicate.matches(TupleDomainParquetPredicate.java:111)
	at com.facebook.presto.hive.parquet.predicate.ParquetPredicateUtils.predicateMatches(ParquetPredicateUtils.java:121)
	at com.facebook.presto.hive.parquet.ParquetPageSourceFactory.lambda$createParquetPageSource$2(ParquetPageSourceFactory.java:188)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:189)
	... 16 more
```

### Query used

```
select
  cd_gender, cd_marital_status, cd_education_status, count(*) cnt1,
  cd_purchase_estimate, count(*) cnt2, cd_credit_rating, count(*) cnt3,
  cd_dep_count, count(*) cnt4, cd_dep_employed_count,  count(*) cnt5,
  cd_dep_college_count, count(*) cnt6
 from
  customer c, customer_address ca, customer_demographics
 where
  c.c_current_addr_sk = ca.ca_address_sk and
  ca_county in ('Rush County','Toole County','Jefferson County',
                'Dona Ana County','La Porte County') and
  cd_demo_sk = c.c_current_cdemo_sk AND
  exists (select * from store_sales, date_dim
          where c.c_customer_sk = ss_customer_sk AND
                ss_sold_date_sk = d_date_sk AND
                d_year = 2002 AND
                d_moy between 1 AND 1+3) AND
   (exists (select * from web_sales, date_dim
            where c.c_customer_sk = ws_bill_customer_sk AND
                  ws_sold_date_sk = d_date_sk AND
                  d_year = 2002 AND
                  d_moy between 1 AND 1+3) or
    exists (select * from catalog_sales, date_dim
            where c.c_customer_sk = cs_ship_customer_sk AND
                  cs_sold_date_sk = d_date_sk AND
                  d_year = 2002 AND
                  d_moy between 1 AND 1+3))
 group by cd_gender,
          cd_marital_status,
          cd_education_status,
          cd_purchase_estimate,
          cd_credit_rating,
          cd_dep_count,
          cd_dep_employed_count,
          cd_dep_college_count
 order by cd_gender,
          cd_marital_status,
          cd_education_status,
          cd_purchase_estimate,
          cd_credit_rating,
          cd_dep_count,
          cd_dep_employed_count,
          cd_dep_college_count
LIMIT 100;
```

### Used in conjunction with
 https://github.com/databricks/benchmarks/tree/master/tpc-ds-2.4/emr-presto

This is using the test data from https://github.com/databricks/benchmarks

Presto verison `presto-server-0.205`

**Comment by tooptoop4** [comment]
presto:dh_si> select type_code from hive.DH_SI.mdm_mv_cd_reference where PART_DT = '20180309' and PART_SEQ = '1' and code_table_name = 'CDADMINFLDNMTP';

Query 20180706_064443_00164_bfmmb, FAILED, 1 node
Splits: 56 total, 0 done (0.00%)
0:01 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20180706_064443_00164_bfmmb failed: Error opening Hive split s3a://redacted/red1/red2/red3/mdm/output/MDM_MV_CD_REFERENCE/part_dt=20180309/part_seq=1/part-00003-1dcb97fd-8ffe-4724-8b05-746e0cb00401.snappy.parquet (offset=0, length=11420): Mismatched Domain types: varchar(21) vs varchar

presto:dh_si>



DDL for reference:

presto:dh_si> show create table hive.DH_SI.mdm_mv_cd_reference;
                                                                         Create Table
---------------------------------------------------------------------------------------------------------------------------------------------------------------
 CREATE TABLE hive.dh_si.mdm_mv_cd_reference (
    code_table_name varchar(21),
    type_code bigint,
    type_description varchar(24),
    parent_type_code varchar(50),
    parent_type_description varchar(16),
    name varchar(120),
    description varchar(255),
    expiry_dt timestamp,
    metadata__job_id varchar(50),
    part_dt varchar,
    part_seq varchar
 )
 WITH (
    external_location = 's3a://redacted/red1/red2/red3/mdm/output/MDM_MV_CD_REFERENCE',
    format = 'PARQUET',
    partitioned_by = ARRAY['part_dt','part_seq']
 )
(1 row)

**Comment by tooptoop4** [comment]
can this be included in .206 release? its real blocker for us as essentially means presto can't read parquet data from hive. Similar issues: #9975 , #9084

**Comment by findepi** [comment]
cc @zhenxiao

**Comment by nezihyigitbasi** [comment]
@antw12 @tooptoop4 #9975 has been merged, and it should fix this problem. The fix will go out in the next release (0.207). Please give that a try and file an issue if the problem still persists.
